mlflow/store/file_store.py

```diff
@@ -235,12 +235,18 @@ def rename_experiment(self, experiment_id, new_name):
 
     def delete_run(self, run_id):
         run_info = self._get_run_info(run_id)
+        if run_info is None:
+            raise MlflowException("Run '%s' not found" % run_id,
```

Comment:
Hm since we'll only reach this case if we find a corrupt run (and never if the run is actually missing, since we check that [here](https://github.com/mlflow/mlflow/pull/661/files#diff-73334c51ec814d7e8b40d09d7d828f82R348)), maybe our exception / error should actually reflect that? For example, we could `raise MlflowException("Run '%s' was malformed" % run_id, databricks_pb2.INVALID_STATE)`